app/js/postEditFunction.ts

```diff
@@ -10,8 +10,11 @@ import {BaseUser} from "./interfaces/User";
  */
 function createObjectForDatabase(inputClass: string) {
     let formData = document.querySelectorAll(inputClass);
-    let result = {'canRetake': document.querySelector<HTMLInputElement>('#canRetake').checked};
-
+    if (document.querySelector<HTMLInputElement>('#canRetake').checked) {
+        var result = {'canRetake': 1}
+    } else {
+         result = {'canRetake': 0}
+    }
```

Comment:
This works well but can be simplified a bit!

Since the problem is that the original result object was just populated with the value of the checkbox directly, you could wrap it in a ternary instead of the if statement.

`let result = {'canRetake': (document.querySelector<HTMLInputElement>('#canRetake').checked ? 1 : 0)};`

So if the checked value is true, the ternary uses a value of 1, and if it's false it uses 0.

Has the same effect as your version but is just a bit simpler!